### DISCUSSION TITLE
Possible vulnerability and inconsistency of comments with the actual code

### DISCUSSION THREAD

**Discussion by SeBun** [joomla/joomla-cms#43065] [General]
Please check these files in 4 and 5 releases.
\libraries\vendor\voku\portable-ascii\src\voku\helper\ASCII.php
\libraries\vendor\voku\portable-utf8\src\voku\helper\UTF8.php
Javascript variables are also obfuscated in the files. 
https://github.com/voku/portable-utf8/blob/master/CHANGELOG.md#604-2022-03-08

**Comment by progreccor** [comment]
I see here  Javascript variable obfuscation and maybe vendor was hacked? or  has a malicious intent to embed a backdoor in the code. 
maybe security problem with  this vendor.
Need to be checked by Security team.

**Comment by Sulpher** [comment]
Joomla community stays outside of political debates.
As far as I understood, this is an official position of library's developer. In such a way, a fork of the library would be a good solution.

**Comment by deoverus** [comment]
Let me make a request regarding the exclusion of the possibility of displaying political statements in the CMS code. Our community has always been apolitical, and to maintain this principle we ask that changes be made accordingly. Given the growing threat of CMS hacking, we are concerned about the security of our platform and ask for your attention and support.

**Comment by rinka88** [comment]
Joomla CMS should stay out of politics. And by this all third-party connections in the kernel should preferably be checked and not to make a mistake. 
I think the security team should check all third-party connections in the kernel for policy.

**Comment by jwebru** [comment]
How can you vote for “the best CMS of the year” after this?

**Comment by CB9TOIIIA** [comment]
Maybe security problem with this vendor, **need to be checked by Security team.**

**Comment by bembelimen** [comment]
I don't see any difference between shipped code and code in GitHub.

**Reply by progreccor** [reply]
code has Javascript variable obfuscation and has comments that have nothing to do with the code

**Reply by Septdir** [reply]
https://github.com/voku/portable-utf8/blob/master/src/voku/helper/UTF8.php#L7
this is a big problem
